google/cloud/internal/opentelemetry.h

```diff
@@ -83,6 +83,36 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
                                     opentelemetry::common::AttributeValue>>
         attributes);
 
+/**
+ * Start a [span] using the current [tracer] with the specified attributes and
+ * links.
+ *
+ * The current tracer is determined by the prevailing `CurrentOptions()`.
+ *
+ * @see https://opentelemetry.io/docs/instrumentation/cpp/manual/#start-a-span
+ *
+ * [span]:
+ * https://opentelemetry.io/docs/concepts/signals/traces/#spans-in-opentelemetry
+ * [tracer]: https://opentelemetry.io/docs/concepts/signals/traces/#tracer
+ */
+template <class T>
+opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
+    opentelemetry::nostd::string_view name,
+    std::initializer_list<std::pair<opentelemetry::nostd::string_view,
+                                    opentelemetry::common::AttributeValue>>
+        attributes,
+    T const& links) {
```

Comment:
Can we use this class instead of a template?

https://github.com/open-telemetry/opentelemetry-cpp/blob/18a27df365e0c347bd54db47341575e6b232dba0/sdk/src/trace/tracer.cc#L25

And is it possible to combine these functions using defaulted parameters?

Reply:
Done

I followed what Otel does...